test/e2e/autoscale_test.go

```diff
@@ -200,56 +202,66 @@ func assertScaleDown(ctx *testContext) {
 	ctx.t.Log("Scaled down.")
 }
 
-func TestAutoscaleUpDownUp(t *testing.T) {
-	t.Parallel()
-	ctx := setup(t)
-	stopChan := DiagnoseMeEvery(t, 15*time.Second, ctx.clients)
-	defer close(stopChan)
-	defer test.TearDown(ctx.clients, ctx.names)
-
-	assertScaleUp(ctx)
-	assertScaleDown(ctx)
-	assertScaleUp(ctx)
-}
-
-func assertNumberOfPods(ctx *testContext, numReplicasMin int32, numReplicasMax int32) error {
-	deployment, err := ctx.clients.KubeClient.Kube.AppsV1().Deployments(test.ServingNamespace).Get(ctx.deploymentName, metav1.GetOptions{})
+func numberOfPods(ctx *testContext) (int32, error) {
+	deployment, err := ctx.clients.KubeClient.Kube.Apps().Deployments(test.ServingNamespace).Get(ctx.deploymentName, metav1.GetOptions{})
 	if err != nil {
-		return errors.Wrapf(err, "Failed to get deployment %q", deployment)
-	}
-	gotReplicas := deployment.Status.ReadyReplicas
-	mes := fmt.Sprintf("got %d replicas, expected between [%d, %d] replicas for deployment %s", gotReplicas, numReplicasMin, numReplicasMax, ctx.deploymentName)
-	ctx.t.Log(mes)
-	if gotReplicas < numReplicasMin || gotReplicas > numReplicasMax {
-		return errors.New(mes)
+		return 0, errors.Wrapf(err, "Failed to get deployment %q", deployment)
 	}
-	return nil
+	return deployment.Status.ReadyReplicas, nil
 }
 
-func assertAutoscaleUpToNumPods(ctx *testContext, numPods int32) {
-	// Relaxing the pod count requirement a little bit to avoid being too flaky.
-	minPods := numPods - 1
-	maxPods := numPods + 1
+func assertAutoscaleUpToNumPods(ctx *testContext, curPods, targetPods int32, duration time.Duration, quick bool) {
+	// There are two test modes: quick, and not quick.
+	// 1) Quick mode: succeeds when the number of pods meets targetPods.
+	// 2) Not Quick (sustaining) mode: succeeds when the number of pods gets scaled to targetPods and
+	//    sustains there for the `duration`.
+
+	minPods := curPods
+	// Leave ample headroom because when switching to panic mode, the app is running at
+	// panicThresholdPercentage of the containerConcurrency (and panicThresholdPercentage is at least 110).
```

Comment:
we should just use the number we're utilizing in the test, since this is more-or-less precise computation with steady traffic.

Reply:
The value is already good (the value is not set and defaulted to 200 in configmap).